### PR TITLE
Default Theme

### PR DESCRIPTION
**Pull request by adamhstewart**
#4

### CODE REVIEW

**Comment by adamhstewart** [comment]
image

image

**Comment by adamhstewart** [comment]
@amitaibu Still to do:

- Make sourcemaps work correctly
- Compile JS in dist folder

those can be followups.

File: .ddev/config.local.yaml.example [hunk 1]
```diff
@@ -13,4 +13,5 @@ hooks:
 
     # Install profile
     - exec: drush site-install server -y --existing-config
+    - exec: cd .. && ./vendor/bin/robo compile:theme
```
Comment (amitaibu) [inline .ddev/config.local.yaml.example:16]:
```suggestion

    # Compile theme
    - exec-host: ddev robo compile:theme
```

File: RoboFile.php [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+          },
+          FilesystemEvent::ALL
+        )->run();
+    }
+  }
+
+  /**
+   * Watch the theme path and compile on change.
+   *
+   * Non-optimized, for `Debug.toString`.
```
Comment (amitaibu) [inline RoboFile.php:124]:
```suggestion
   * Watch the theme path and compile on change (non-optimized).
```

File: web/themes/custom/theme_server/theme_server.libraries.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    theme:
+      dist/css/style.css: {}
+
+global-scripts:
+  js:
+    js/scripts.js: {}
+  dependencies:
+    - core/jquery
+librefranklin:
+  css:
```
Comment (amitaibu) [inline web/themes/custom/theme_server/theme_server.libraries.yml:13]:
```suggestion
    - core/jquery

librefranklin:
  css:
```

File: web/themes/custom/theme_server/theme_server.libraries.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+global-styling:
+  version: 1.0
+  css:
+    theme:
+      dist/css/style.css: {}
+
+global-scripts:
+  js:
+    js/scripts.js: {}
```
Comment (amitaibu) [inline web/themes/custom/theme_server/theme_server.libraries.yml:9]:
```suggestion
    src/js/scripts.js: {}
```

But:

1. Lets comment out this section, so it doesn't load that JS file.

2. Lets add to scripts.js

```
/**
 * @file
 */

(function ($, Drupal) {

  /**
   * Your script goes here.
   */
  Drupal.behaviors.themeServerGeneral = {
    attach: function (context, settings) {
    }
  };
})(jQuery, Drupal);

```

File: RoboFile.php [hunk 5]
```diff
@@ -0,0 +1,141 @@
+<?php
+
+use ScssPhp\ScssPhp\Compiler;
+use Lurker\Event\FilesystemEvent;
+use Robo\Tasks;
+use Symfony\Component\EventDispatcher\Event;
+
+/**
+ * Theme compilation Robo file.
```
Comment (amitaibu) [inline RoboFile.php:9]:
```suggestion
 * Robo commands.
```

File: RoboFile.php [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+      $this->taskImageMinify(self::THEME_BASE . '/src/images/*.png')
+        ->to(self::THEME_BASE . '/dist/images/')
+        ->run();
+    }
+  }
+
+  /**
+   * Compile the theme (optimized).
+   */
+  public function compileTheme() {
```
Comment (amitaibu) [inline RoboFile.php:76]: IMO it should be `theme:compile`, `theme:watch`, `theme:watch:debug` - so we should change names to `themeCompile`, `themeWatch`, `themeWatchDebug`

**Comment by adamhstewart** [comment]
@amitaibu  review comments addressed - I added JS minification, but not sure it's exactly working correctly. It is at least copying the script files into the `dist` directory.

**Comment by amitaibu** [comment]
Thanks, I've fixed JS minify.

**Pull request merged**